in_toto/runlib.go

```diff
@@ -48,6 +49,35 @@ func RecordArtifact(path string) (map[string]interface{}, error) {
 	return hashedContentsMap, nil
 }
 
+// ApplyExcludePatterns takes paths and excludePatterns as input
+// and compares both. If the excludePattern matches a file path
+// the file will not be included.
+// This behavior is similar to git's famous gitignore file.
+func ApplyExcludePatterns(paths []string, excludePatterns []string) ([]string, error) {
+	if excludePatterns == nil {
+		return paths, nil
+	}
+	// parses the exclude patterns into pattern type object
+	var ps []Pattern
+	var included []string
+	for _, element := range excludePatterns {
+		temp := ParsePattern(element, nil)
+		ps = append(ps, temp)
+	}
+	m := NewMatcher(ps)
+	for _, path := range paths {
+		fileInfo, err := os.Stat(path)
```

Comment:
No, because we are dealing with a gitignore matcher here. This matcher needs to know if we are dealing with a directory or not: https://pkg.go.dev/gopkg.in/src-d/go-git.v4/plumbing/format/gitignore?tab=doc#Matcher

That's why we need FileInfo to retrieve information about the actual state of the file (directory or not).

Reply:
Thanks for the clarification! This also answers my question below. Would you mind adding a comment?

Reply:
yep